presto-hive-metastore/src/main/java/com/facebook/presto/hive/authentication/HiveMetastoreAuthentication.java

```diff
@@ -18,4 +18,17 @@
 public interface HiveMetastoreAuthentication
 {
     TTransport authenticate(TTransport rawTransport, String hiveMetastoreHost);
+
+    TTransport authenticateWithToken(TTransport rawTransport, String tokenString);
+
+    <R, E extends Exception> R doAs(String user, GenericExceptionAction<R, E> action)
+            throws E;
+
+    default void doAs(String user, Runnable action)
+    {
+        doAs(user, () -> {
+            action.run();
+            return null;
```

Comment:
do we need to return null?

Reply:
yes, because action.run() is void and doAs needs to return